### ISSUE TITLE
Question in SatrnEncoder

### ISSUE THREAD

**Issue by YanShuang17** [open-mmlab/mmocr#649]
Line 71 in `/mmocr/models/textrecog/encoders/satrn_encoder.py`:

```python
        feat += self.position_enc(feat)
```

I noticed that the `Add` operation between `feat` and `position_enc` vecter has been executed in `forward()` func of Class `Adaptive2DPositionalEncoding`(see line 163 in `mmocr/models/textrecog/layers/satrn_layers.py`).

So it should be like this ?

```python
        feat = self.position_enc(feat)
```

Although it has little influence...

**Comment by gaotongxiao** [comment]
You're right, that was a mistake. You are welcome to raise a PR to fix this issue, which will take a longer time to merge since we will have to re-train the model based on the new architecture.

**Comment by yellowjs0304** [comment]
Hi, @gaotongxiao @YanShuang17  I'm confused that your issue.
I'm not sure, but It looks like right that you coded before.

Here's the[Clova's SATRN(origin repo)](https://github.com/clovaai/SATRN/blob/d4c257b2732137858827041f81761962a27f5228/src/networks/SATRN.py#L230), 

I think it looks same with before code `feat+= self.position_enc(feat)`.
Could you please check about this?

**Comment by yellowjs0304** [comment]
Now, I understand In your [repo ](https://github.com/open-mmlab/mmocr/blob/c6c4acd39b1c56fec1b87530b2d241fe8af4ceed/mmocr/models/textrecog/layers/satrn_layers.py#L163), it looks like you already summed the input tensor and positional tensor, right?

If this right, then the code should be changed as `feat = self.position_enc(feat)`

**Comment by gaotongxiao** [comment]
@yellowjs0304 Yep, there's a minor implementation difference. We will fix it but it takes time to retrain SATRN. For now, we can append a note to this line to warn others.

**Comment by gaotongxiao** [comment]
It's fixed in MMOCR 1.0.
